### DISCUSSION TITLE
Problem with calling through the auto1111 API. Please help.

### DISCUSSION THREAD

**Discussion by DennLie** [thygate/stable-diffusion-webui-depthmap-script#110] [Q&A]
I am trying to create a depthmap in unity by calling this awesome script with the 28 arguments needed in the def run() method. The call is from /sdapi/v1/img2img . I got Img2Img working with other custom scripts already so i think i the error is probably by wrong datatype declaration of the script argument variables. It would be awesome if someone with bigger knowledge could take a quick look and tell me which datatypes are not correct. For example i tried ushort, float, double with clipthreshold_far and clipthreshold_near but that didnt do the trick.
<img width="475" alt="1" src="https://user-images.githubusercontent.com/50552332/218168935-e054e5d2-abee-46f4-bc43-93f0dcbbc871.png">
<img width="944" alt="2" src="https://user-images.githubusercontent.com/50552332/218168950-f1f3fa6a-1545-4d24-a8cc-54fe80b9e168.png">
<img width="228" alt="3" src="https://user-images.githubusercontent.com/50552332/218168962-0b11fcc3-85ac-4dfa-ae77-5aedf9c7052a.png">

**Comment by DennLie** [comment]
Ok found my error :)

**Comment by aulerius** [comment]
Are you still using it through the API like that?
Could you post the list of arguments and their types that worked for you?

**Reply by DennLie** [reply]
Hey, i changed so much and i am not using this script anymore through API. I took a look in my scripts but seems like that part is gone completely. If i remember correctly i was using a wrong datatype in one of the arguments.
 If you are not familiar with this handy little extension  take a look here: 
 https://github.com/huchenlei/sd-webui-api-payload-display  
 With this you can see the API payload after each generation on WebUI. The naming for arguments in custom scripts doesnt work  and is shown as 1,2,3 but as they are the arguments in the def run() it is easy to rename them properly. No need to guess the right datatype anymore.
Wish i would have found that earlier myself...
